### PR TITLE
 #22 allow custom variables in migrations

### PR DESCRIPTION
**Pull request by larochef**
As stated in #22 I'd need the ability to pass variables to my migration script, and here is a MR that would allow it.

What do you think of it?

### CODE REVIEW

File: src/main/java/com/github/couchmove/Couchmove.java [hunk 1]
```diff
@@ -289,8 +312,8 @@ void doExecute(ChangeLog changeLog) {
             if (type == DOCUMENTS) {
                 dbService.importDocuments(fileService.readDocuments(changeLog.getScript()));
             } else {
-                var description = changeLog.getDescription().replace(" ", "_");
-                var content = fileService.readFile(changeLog.getScript());
+                String description = changeLog.getDescription().replace(" ", "_");
```
Comment (tchlyah) [inline src/main/java/com/github/couchmove/Couchmove.java:315]: This is not related to this PR

File: src/main/java/com/github/couchmove/repository/CouchbaseRepositoryImpl.java [hunk 2]
```diff
@@ -251,7 +265,7 @@ public void importEventingFunctions(String description, String jsonContent) {
 
         logger.debug("Import {} Eventing Functions", functions.length);
         for (int i = 0; i < functions.length; i++) {
-            var function = functions[i];
+            EventingFunction function = functions[i];
```
Comment (tchlyah) [inline src/main/java/com/github/couchmove/repository/CouchbaseRepositoryImpl.java:268]: This is not related to this PR

File: src/main/java/com/github/couchmove/repository/CouchbaseRepositoryImpl.java [hunk 3]
```diff
@@ -332,8 +343,8 @@ public void watchN1qlIndexes(Duration duration) {
 
     @Override
     public void watchN1qlIndexes(String scope, String collection, Duration duration) {
-        var getAllIndexesOptions = withRetry(getAllQueryIndexesOptions());
-        var watchQueryIndexesOptions = watchQueryIndexesOptions();
+        GetAllQueryIndexesOptions getAllIndexesOptions = withRetry(getAllQueryIndexesOptions());
```
Comment (tchlyah) [inline src/main/java/com/github/couchmove/repository/CouchbaseRepositoryImpl.java:346]: This is not related to this PR

File: src/main/java/com/github/couchmove/service/ChangeLockService.java [hunk 4]
```diff
@@ -31,11 +32,11 @@ public class ChangeLockService {
     private String uuid;
 
     public ChangeLockService(Bucket bucket, Cluster cluster) {
-        this.repository = new CouchbaseRepositoryImpl<>(cluster, bucket, ChangeLock.class);
+        this.repository = new CouchbaseRepositoryImpl<>(cluster, bucket, ChangeLock.class, Collections.emptyMap());
```
Comment (tchlyah) [inline src/main/java/com/github/couchmove/service/ChangeLockService.java:35]: This change can be avoided by adding new constructor without variables

File: src/main/java/com/github/couchmove/utils/Utils.java [hunk 5]
```diff
@@ -44,7 +44,7 @@ private static String initializeUserName() {
             try {
                 Class<?> c = Class.forName(className);
                 Method method = c.getDeclaredMethod(methodName);
-                Object o = c.newInstance();
+                Object o = c.getDeclaredConstructor().newInstance();
```
Comment (tchlyah) [inline src/main/java/com/github/couchmove/utils/Utils.java:47]: Can you explain what this change is for ?

File: src/main/java/com/github/couchmove/service/ChangeLogDBService.java [hunk 6]
```diff
@@ -135,7 +135,7 @@ public void executeN1ql(String content) {
     public void importDocuments(Collection<Document> documents) {
         logger.info("Importing {} documents", documents.size());
         for (Document document : documents) {
-            var repo = repository;
+            CouchbaseRepository<ChangeLog> repo = repository;
```
Comment (tchlyah) [inline src/main/java/com/github/couchmove/service/ChangeLogDBService.java:138]: This is not related to this PR

File: src/test/java/com/github/couchmove/service/ChangeLogFileServiceTest.java [hunk 7]
```diff
@@ -110,7 +110,7 @@ public void should_fetch_changeLogs() throws IOException {
     @Test
     public void should_fetch_files_with_same_version() throws IOException {
         // Given a temp directory that contains
-        File tempDir = Files.createTempDir();
+        File tempDir = java.nio.file.Files.createTempDirectory("couchmove").toFile();
```
Comment (tchlyah) [inline src/test/java/com/github/couchmove/service/ChangeLogFileServiceTest.java:113]: This is not related to this PR

File: src/test/java/com/github/couchmove/service/ChangeLogFileServiceTest.java [hunk 8]
```diff
@@ -123,7 +123,7 @@ public void should_fetch_files_with_same_version() throws IOException {
         Files.write(content2.getBytes(), file2);
 
         // When we fetch changelogs in this directory
-        var results = new ChangeLogFileService(tempDir.toPath().toString()).fetch();
+        List<ChangeLog> results = new ChangeLogFileService(tempDir.toPath().toString()).fetch();
```
Comment (tchlyah) [inline src/test/java/com/github/couchmove/service/ChangeLogFileServiceTest.java:126]: This is not related to this PR

File: src/test/java/com/github/couchmove/utils/FileUtilsTest.java [hunk 9]
```diff
@@ -78,7 +78,7 @@ public void should_read_files_failed_if_not_directory() throws Exception {
     @Test
     public void should_read_files_in_directory() throws IOException {
         // Given a temp directory that contains
-        File tempDir = Files.createTempDir();
+        File tempDir = java.nio.file.Files.createTempDirectory("couchmove").toFile();
```
Comment (tchlyah) [inline src/test/java/com/github/couchmove/utils/FileUtilsTest.java:81]: This is not related to this PR

File: src/test/java/com/github/couchmove/utils/FileUtilsTest.java [hunk 10]
```diff
@@ -91,9 +91,9 @@ public void should_read_files_in_directory() throws IOException {
         // txt file
         Files.write(getRandomString().getBytes(), File.createTempFile(getRandomString(), ".txt", tempDir));
         // json file with scope/collection directories
-        var scope = "scope";
-        var collection = "collection";
-        var collectionDir = new File(tempDir, format("%s/%s", scope, collection));
+        String scope = "scope";
```
Comment (tchlyah) [inline src/test/java/com/github/couchmove/utils/FileUtilsTest.java:94]: This is not related to this PR

File: src/test/java/com/github/couchmove/utils/FileUtilsTest.java [hunk 11]
```diff
@@ -106,7 +106,7 @@ public void should_read_files_in_directory() throws IOException {
         Files.write(content4.getBytes(), file4);
 
         // When we read files in this directory with extension filter
-        var results = FileUtils.readFilesInDirectory(tempDir.toPath(), "json", "n1ql");
+        Collection<Document> results = FileUtils.readFilesInDirectory(tempDir.toPath(), "json", "n1ql");
```
Comment (tchlyah) [inline src/test/java/com/github/couchmove/utils/FileUtilsTest.java:109]: This is not related to this PR

File: src/test/java/com/github/couchmove/CouchmoveIT.java [hunk 12]
```diff
@@ -53,9 +53,9 @@ public class CouchmoveIT extends BaseIT {
 
     @BeforeEach
     public void init() {
-        changeLogRepository = new CouchbaseRepositoryImpl<>(getCluster(), getBucket(), ChangeLog.class);
-        changeLogDBService = new ChangeLogDBService(getBucket(), getCluster());
-        userRepository = new CouchbaseRepositoryImpl<>(getCluster(), getBucket(), User.class);
+        changeLogRepository = new CouchbaseRepositoryImpl<>(getCluster(), getBucket(), ChangeLog.class, Collections.emptyMap());
```
Comment (tchlyah) [inline src/test/java/com/github/couchmove/CouchmoveIT.java:56]: This change can be avoided by adding new constructor without variables

File: src/test/java/com/github/couchmove/CouchmoveIT.java [hunk 13]
```diff
@@ -432,7 +432,7 @@ public void should_insert_changelogs_into_collection() {
         couchmove.migrate();
 
         // Then all changeLogs should be inserted in the same collection
-        var collectionChangeLogRepository = changeLogRepository.withCollection(CHANGELOG_SCOPE, CHANGELOG_COLLECTION);
+        CouchbaseRepository<ChangeLog> collectionChangeLogRepository = changeLogRepository.withCollection(CHANGELOG_SCOPE, CHANGELOG_COLLECTION);
```
Comment (tchlyah) [inline src/test/java/com/github/couchmove/CouchmoveIT.java:435]: This is not related to this PR

File: src/test/java/com/github/couchmove/CouchmoveIT.java [hunk 14]
```diff
@@ -415,7 +415,7 @@ public void should_insert_changelogs_into_collection() {
         Collection collection = getBucket().scope(CHANGELOG_SCOPE).collection(CHANGELOG_COLLECTION);
 
         // And a Couchmove instance configured with this collection
-        var couchmove = new Couchmove(collection, getCluster(), DB_MIGRATION + "collections");
+        Couchmove couchmove = new Couchmove(collection, getCluster(), DB_MIGRATION + "collections");
```
Comment (tchlyah) [inline src/test/java/com/github/couchmove/CouchmoveIT.java:418]: This is not related to this PR

**Review by tchlyah — CHANGES_REQUESTED** [review]

File: src/main/java/com/github/couchmove/utils/Utils.java (hunk 5 above)
Comment (larochef) [reply src/main/java/com/github/couchmove/utils/Utils.java:47]: It  was just removing deprecation warnings (like for the vars), it is deprecated since 9.
I can remove it since it's not related to the PR

File: src/main/java/com/github/couchmove/Couchmove.java (hunk 1 above)
Comment (larochef) [reply src/main/java/com/github/couchmove/Couchmove.java]: removed

File: src/main/java/com/github/couchmove/repository/CouchbaseRepositoryImpl.java (hunk 2 above)
Comment (larochef) [reply src/main/java/com/github/couchmove/repository/CouchbaseRepositoryImpl.java]: removed

File: src/main/java/com/github/couchmove/repository/CouchbaseRepositoryImpl.java (hunk 3 above)
Comment (larochef) [reply src/main/java/com/github/couchmove/repository/CouchbaseRepositoryImpl.java]: removed

File: src/main/java/com/github/couchmove/service/ChangeLockService.java (hunk 4 above)
Comment (larochef) [reply src/main/java/com/github/couchmove/service/ChangeLockService.java]: Done

File: src/test/java/com/github/couchmove/utils/FileUtilsTest.java (hunk 10 above)
Comment (larochef) [reply src/test/java/com/github/couchmove/utils/FileUtilsTest.java]: removed

File: src/test/java/com/github/couchmove/utils/FileUtilsTest.java (hunk 11 above)
Comment (larochef) [reply src/test/java/com/github/couchmove/utils/FileUtilsTest.java]: removed

File: src/test/java/com/github/couchmove/CouchmoveIT.java (hunk 12 above)
Comment (larochef) [reply src/test/java/com/github/couchmove/CouchmoveIT.java]: Done

File: src/test/java/com/github/couchmove/CouchmoveIT.java (hunk 13 above)
Comment (larochef) [reply src/test/java/com/github/couchmove/CouchmoveIT.java]: removed

File: src/test/java/com/github/couchmove/CouchmoveIT.java (hunk 14 above)
Comment (larochef) [reply src/test/java/com/github/couchmove/CouchmoveIT.java]: removed

**Comment by larochef** [comment]
@tchlyah  All should be ok now, can you check if it fit what you are expecting?
